### PR TITLE
Admin can change user's donor level

### PR DESCRIPTION
**Pull request by anoshaminai**
- [x] Added dropdown with donor levels next to every user in the 'registered user' list on the admin dashboard
- [x] Default dropdown value is filled with current user donor level (from database)
- [x] TO DO: Update db with selected donor level
- [x] TO DO: Get donor level choices from the db, instead of hard-coding
- [x] TO DO: Make default value dark text, not greyed out 

dropdown1

### CODE REVIEW

File: app/templates/admin/registered_users.html [hunk 1]
```diff
@@ -1,67 +1,77 @@
...
 
 {% block content %}
-    <div class="ui stackable grid container">
-        <div class="sixteen wide tablet twelve wide computer centered column">
-            <a class="ui basic compact button" href="{{ url_for('admin.index') }}">
-                <i class="caret left icon"></i>
-                Back to dashboard
-            </a>
-            <h2 class="ui header">
+<div class="ui stackable grid container">
```
Comment (aharelick) [inline app/templates/admin/registered_users.html]: this should all be indented one over from the `{% block content %}`

**Comment by aharelick** [comment]
@anoshaminai: it's hard to tell what was changed because the indentation was changed for a lot of the html. Take a look at https://github.com/hack4impact/women-veterans-rock/pull/27/files to see what I'm talking about. It'll be a lot easier to see the differences when things are properly indented again.

**Comment by anoshaminai** [comment]
@aharelick Is that the correct indentation?

**Comment by anoshaminai** [comment]
@aharelick @Mebsworth Please take a look !

**Comment by aharelick** [comment]
@anoshaminai, I don't think you ever pushed your most recent code, looks like the most recent commit is from awhile ago

**Comment by anoshaminai** [comment]
@aharelick Sorry about that, should all be there now!

File: app/admin/views.py [hunk 2]
```diff
@@ -159,3 +160,21 @@ def delete_user(user_id):
...
+
+
+@admin.route('/users/update-donor-level', methods=['POST'])
+@login_required
+@admin_required
+def update_donor_level():
+    user_id = request.form.get('user_id')
+    dl_name = request.form.get('donor_level')
+
+    user = User.query.filter_by(id=user_id).first()
```
Comment (aharelick) [inline app/admin/views.py]: `user = User.query.get(user_id)`

Comment (aharelick) [inline app/admin/views.py]: there should also be some `if user is None:` check here

File: app/admin/views.py [hunk 3]
```diff
@@ -159,3 +160,21 @@ def delete_user(user_id):
...
+    donor_level = DonorLevel.query.filter_by(name=dl_name).first()
+    dl_id = donor_level.id
+    user.donor_level_id = dl_id
```
Comment (aharelick) [inline app/admin/views.py]: `user.donor_level_id = donor_level.id`, just combine it into one line

File: app/admin/views.py [hunk 4]
```diff
@@ -159,3 +160,21 @@ def delete_user(user_id):
...
+
+@admin.route('/users/update-donor-level', methods=['POST'])
+@login_required
+@admin_required
+def update_donor_level():
+    user_id = request.form.get('user_id')
+    dl_name = request.form.get('donor_level')
+
+    user = User.query.filter_by(id=user_id).first()
+    donor_level = DonorLevel.query.filter_by(name=dl_name).first()
```
Comment (aharelick) [inline app/admin/views.py]: another check here to make sure the `donor_level` is not None

File: app/models/user.py [hunk 5]
```diff
@@ -64,6 +64,20 @@ class DonorLevel(db.Model):
...
     users = db.relationship('User', backref='donor_level', lazy='dynamic')
 
+    @staticmethod
+    def insert_donor_levels():
+        donor_levels = {
+            'Advocate',
+            'Supporter',
+            'Benefactor'
+        }
+        for dl in donor_levels:
```
Comment (aharelick) [inline app/models/user.py]: let's use a different variable that's not `dl`. It's kind of confusing with the `donor_level` one below

File: app/templates/admin/registered_users.html [hunk 6]
```diff
@@ -35,8 +33,7 @@ <h2 class="ui header">
                 </div>
             </div>
 
-            {# Use overflow-x: scroll so that mobile views don't freak out
-             # when the table is too wide #}
+            {# Use overflow-x: scroll so that mobile views don't freak out # when the table is too wide #}
```
Comment (aharelick) [inline app/templates/admin/registered_users.html]: take out the `#` just chillin in the there

File: app/templates/admin/registered_users.html [hunk 7]
```diff
@@ -45,24 +42,37 @@ <h2 class="ui header">
...
+                                </td>
+                            </tr>
+                        {% endfor %}
                     </tbody>
                 </table>
             </div>
         </div>
     </div>
 
+
```
Comment (aharelick) [inline app/templates/admin/registered_users.html]: don't need this new line

File: app/templates/admin/registered_users.html [hunk 8]
```diff
@@ -84,6 +94,22 @@ <h2 class="ui header">
...
+                        donor_level: value,
+                        csrf_token: "{{ csrf_token() }}"
+                    };
+                    $.post("{{ url_for('admin.update_donor_level') }}", donor_level_json_obj)
+                }
+             });
+
+
         });
+
```
Comment (aharelick) [inline app/templates/admin/registered_users.html]: remove all these empty lines

File: app/templates/admin/registered_users.html [hunk 9]
```diff
@@ -84,6 +94,22 @@ <h2 class="ui header">
...
+
+
     </script>
 {% endblock %}
```
Comment (aharelick) [inline app/templates/admin/registered_users.html]: new line here though

File: app/templates/admin/registered_users.html [hunk 10]
```diff
@@ -45,24 +42,37 @@ <h2 class="ui header">
...
-
-                            <td>{{ u.first_name }}</td>
-                            <td>{{ u.last_name }}</td>
-                            <td>{{ u.email }}</td>
-                            <td class="user role">{{ u.role.name }}</td>
-                        </tr>
-                    {% endfor %}
+                        {% for u in users | sort(attribute='last_name') %}
+                            <tr>
+                                <td onclick="window.location.href = '{{ url_for('admin.user_info', user_id=u.id) }}';">{{ u.first_name }}</td>
```
Comment (aharelick) [inline app/templates/admin/registered_users.html]: this is a neat way of doing this, but instead let's make it

``` html
<td><a href="{{ url_for('admin.user_info', user_id=u.id) }}">{{ u.first_name }}</a></td>
```

File: app/templates/admin/registered_users.html [hunk 11]
```diff
@@ -45,24 +42,37 @@ <h2 class="ui header">
...
+                                <td onclick="window.location.href = '{{ url_for('admin.user_info', user_id=u.id) }}';">{{ u.last_name }}</td>
+                                <td onclick="window.location.href = '{{ url_for('admin.user_info', user_id=u.id) }}';">{{ u.email }}</td>
```
Comment (aharelick) [inline app/templates/admin/registered_users.html]: check out the way I did the `<a>` tag above and do the same for the other `<td>` tags

File: app/admin/views.py (hunk 2 above)
Comment (anoshaminai) [inline app/admin/views.py]: is `abort(404)` an appropriate response? it's what is done in several above examples.

also, re: `User.query.get(user_id)` - absolutely, I will make that change that's way cleaner, there are several methods above that use the filter_by stuff I was doing before (several that don't), should those be changed? just a heads up

Comment (aharelick) [inline app/admin/views.py]: sure

**Pull request merged**